### PR TITLE
Introduce pagination for highlight list

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

File: src/ApiClient/HighlightsClient.php [hunk 1]
```diff
@@ -7,12 +7,20 @@
 
 final class HighlightsClient
 {
-    const TYPE_HIGHLIGHTS = 'application/vnd.elife.highlights+json';
+    const TYPE_HIGHLIGHTS = 'application/vnd.elife.highlight-list+json';
```
Comment (thewilkybarkid) [inline src/ApiClient/HighlightsClient.php]: Should become `TYPE_HIGHLIGHT_LIST` for consistency.

File: src/ApiClient/HighlightsClient.php [hunk 2]
```diff
@@ -7,12 +7,20 @@
...
 
     use ApiClient;
 
-    public function list(array $headers, string $id) : PromiseInterface
-    {
-        return $this->getRequest('highlights/'.$id, $headers);
+    public function list(
```
Comment (thewilkybarkid) [inline src/ApiClient/HighlightsClient.php]: Should become `listHighlights()` for consistency.

**Comment by nlisgo** [comment]
@thewilkybarkid I tried that approach but it failed testing because it was trying to call `listHighlights()` without any arguments passed. We need and id.

**Comment by thewilkybarkid** [comment]
Not sure what you mean, just made the change.

**Comment by nlisgo** [comment]
Ignore that then :). I was clearly getting a different error.

**Pull request merged**